### DISCUSSION TITLE
How to import giscus.json in static page with Hugo?

### DISCUSSION THREAD

**Discussion by Haleuan** [giscus/giscus#1585] [Q&A] [answered]
This is a BRILLIANT project!
But I have a question in using giscus.json file. I have my blog built with Hugo and the architecture like the picture:
<img width="331" alt="image" src="https://github.com/user-attachments/assets/51ee3449-f5b9-4424-ad7c-e01b9a2b7a20" />

If I want use giscus in `static/fragments/index.htm`，and use `"defaultCommentOrder": "newest"` attribute. I have tried put giscus.json in `root directory`, `static/`, or `static/fragments`, but all of these method didn't work.

Which directory should I put `giscus.json`?

*Sorry to bother, maybe it's a silly question. But it confused me indeed for a long time...*

**Comment by laymonage** [comment]
Put it in the root of your repo where the discussions exist. The file must be committed and pushed to the GitHub repo.

**Reply by Haleuan** [reply]
I tried, but it still doesn't work😥

Does this method alse works for the single page?
<img width="347" alt="image" src="https://github.com/user-attachments/assets/0947bf39-8796-43f5-94cd-8395f785bd28" />

**Reply by laymonage** [reply]
The repo must be public.

**Reply by Haleuan** [reply, marked as answer]
OH! thank you! problem solved!
